### DISCUSSION TITLE
How to manually sign a file using fastlane/match

### DISCUSSION THREAD

**Discussion by eliaSchenker** [fastlane/fastlane#21982] [Q&A] [answered]
After calling sync_code_signing, how do you sign a binary or file that wasn't built using fastlane (for example a dmg, or a binary executable) with match or fastlane? 

```
platform :macos do
    desc "Signs an existing binary"
    lane :sign do
        sync_code_signing(
            type: "developer_id",
            platform: "macos",
        )
        
        // Sign binary at path ./test-binary with certificate
  end
end
```

Of course a workaround would be to call codesign manually using sh:
```
sh("codesign -s \"Developer ID Application: ...\" ./test-binary")
```

Does a fastlane include a utility for something like that? I checked the docs but couldn't find any.

**Comment by florence0239** [comment, marked as answer]
> After calling sync_code_signing, how do you sign a binary or file that wasn't built using fastlane (for example a dmg, or a binary executable) with match or fastlane?
> 
> ```
> platform :macos do
>     desc "Signs an existing binary"
>     lane :sign do
>         sync_code_signing(
>             type: "developer_id",
>             platform: "macos",
>         )
>         
>         // Sign binary at path ./test-binary with certificate
>   end
> end
> ```
> 
> Of course a workaround would be to call codesign manually using sh:
> 
> ```
> sh("codesign -s \"Developer ID Application: ...\" ./test-binary")
> ```
> 
> Does a fastlane include a utility for something like that? I checked the docs but couldn't find any.

Hello, [MyLowesLife](https://www.my-loweslife.com)
To sign a binary or file that wasn't built using Fastlane (like a dmg or executable), use the sh action in your Fastlane :sign lane after syncing code signing certificates:

`  platform :macos do
  desc "Signs an existing binary"
  lane :sign do
    sync_code_signing(
      type: "developer_id",
      platform: "macos",
    )

    sh("codesign -s \"Developer ID Application: Your Certificate Name\" ./test-binary")
  end
end`

Replace "Developer ID Application: Your Certificate Name" with your actual certificate identifier. This method allows you to manually sign binaries or files outside of Fastlane's build process using the codesign command directly within your automation script.


Best Regard,
florence023
